### PR TITLE
inventory level model

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

fivetran made PR

**What change(s) does this PR introduce?** 

- adds inventory level model and an intermediate model supporting it
- removed duplicative/deprecated columns from order_lines (just to avoid failures for now, will come back to this model in earnest later)
- enhances product model with new tables (I thought product and inventory_level would overlap more in the DAG, but this didn't really happen. consider this an early start on the "enhancing existing models" task)

**Did you update the CHANGELOG?** 

- [ ] Yes
- [ ] nope!! will save it for the documentation task

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [ ] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [X] No but inventory modeling has come up plenty with customers 

**How did you test the PR changes?** 

 

- [x] Buildkite  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
⚡ 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/shopify__products.sql
Comment (fivetran-joemarkiewicz) [inline models/shopify__products.sql:106]: Do you think it would be relevant to include a string agg for the various types of variants? It may not be, but wondering if this would be valuable for customers.

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]: Obligatory reminder to update before merge

File: models/shopify.yml
Comment (fivetran-joemarkiewicz) [reply models/shopify.yml:565]: With how wide these end models are, we definitely want to document the columns as best as we can as well.

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-joemarkiewicz) [inline models/intermediate/shopify__inventory_level__aggregates.sql:20]: With the updates you applied to the [refund temp](https://github.com/fivetran/dbt_shopify_source/blob/feature/package-revamp/models/tmp/stg_shopify__refund_tmp.sql) model, is this conditional needed still?

Comment (fivetran-joemarkiewicz) [reply models/intermediate/shopify__inventory_level__aggregates.sql:20]: It looks like the [order line refund temp](https://github.com/fivetran/dbt_shopify_source/blob/feature/package-revamp/models/tmp/stg_shopify__order_line_refund_tmp.sql) model hasn't gotten the same update. If both receive the update to swap out the variable declaration with the `does exist` logic, I think we can remove the conditionals from these end models altogether.

What are your thoughts?

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-joemarkiewicz) [inline models/intermediate/shopify__inventory_level__aggregates.sql:45]: Will there ever be a situation where both of these are not null? Why should we choose fulfillment location over the order location?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Hey @fivetran-jamie these end models are looking great!! I really like the enhancement we are providing in them and think customers will get a lot of value from them! I have a few change requests and suggestions that should be addressed before this will be good to approve!

- [ ] Address inline comments
- [x] Document the end models in the yml if possible. It will make for easier review and understanding of the models.
- [x] I believe we discussed this and it didn't make sense in the package at the moment - Is there a way we could get a daily inventory model? I could see the inventory levels model being killer if it showed a by day break down. I built one at my last job at it was a fan favorite. That would be awesome if we could do, but I imagine that will be pretty complex. Maybe more of a good community post than a model to start out (I think we chatted about that as well).

**Comment by fivetran-jamie** [comment]
@fivetran-joemarkiewicz yeah so re: daily inventory -- available inventory quantity is only stored as-of-today. a daily model would be possible by taking a _snapshot_ of this model each day. i think it would be worth highlighting that pathway in the README or somewhere, in addition to a community/blog post

**Comment by fivetran-joemarkiewicz** [comment]
> @fivetran-joemarkiewicz yeah so re: daily inventory -- available inventory quantity is only stored as-of-today. a daily model would be possible by taking a _snapshot_ of this model each day. i think it would be worth highlighting that pathway in the README or somewhere, in addition to a community/blog post

Ahhh yes this conversation is coming back to me! Yeah let's do the community/blog post after the fact. Thanks for reminding me!

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql:1]: yeah i was wondering about this -- i noticed the rest of the intermediate models don't have the `int_` prefix, so i was keeping with that naming convention. 

if we were to change the others to `int_*`, this wouldn't _really_ be a breaking change, as the intermediate models are ephemeral (by default at least). what do you think? just prefix this one int model, prefix all of them, or no prefix at all?

File: models/shopify__products.sql
Comment (fivetran-jamie) [reply models/shopify__products.sql:116]: yeah was very much thinking of doing this, just wanted to wait for another pair of eyes to confirm it was necessary/valuable. will move stuff there!

File: models/shopify__products.sql
Comment (fivetran-jamie) [reply models/shopify__products.sql:104]: yeah the thing is a product can have a TON of variants (i've seen 70 variants for 1 product), so a string_agg could get out of hand

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql:45]: in reality, `fulfillment.location_id` won't be null, as order locations are actually stored there. in the data, i have seen `order.location_id` only ever be null, so this is more of a sanity check (but maybe one that is more confusing than helpful?)

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql:20]: are we trying to completely get rid of these variables? because to remove the conditionals, we'd need to remove  `{{ config(enabled=var('shopify__using_refund', True)) }}` from all of the non-tmp refund models, effectively rendering them useless

this sounds good to me, but just wanna confirm!

File: models/shopify.yml
Comment (fivetran-jamie) [reply models/shopify.yml]: added docs!

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-joemarkiewicz) [reply models/intermediate/shopify__inventory_level__aggregates.sql:1]: Looking at the other intermediate models, they seem to have a pretty strange naming convention 🤔. That being said, I am less worried about the breaking change and more worried about the level of effort and the cascading effect of updating these would entail.

Let's stick to our current naming convention for now (naming this with `iint_*`) and then we can open an issue to address the renaming of the models for a future task. That could possibly be a good one for a community contributor to take on if they wanted to contribute to our packages.

File: models/shopify__products.sql
Comment (fivetran-joemarkiewicz) [reply models/shopify__products.sql]: Great insight. Let's leave this for now and if someone ever opens a FR for this type of string agg then we can tackle it then.

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-joemarkiewicz) [reply models/intermediate/shopify__inventory_level__aggregates.sql]: Yeah let's go forward with removing those variables across the board. The refunds and adjustments (to my understanding) are less a feature that is turned off and on and is instead just populated in the source once one comes through. Therefore, the variable doesn't make much sense and would be more appropriate in my opinion to be a default behavior of the package to determine for users.

Let's be sure to document this in the DECISIONLOG so customers are aware of this default behavior and aren't confused by the empty models if they don't have adjustments or refunds.

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-joemarkiewicz) [reply models/intermediate/shopify__inventory_level__aggregates.sql:43]: I guess what is the difference between these two location_ids? I do find this coalesce to be a bit confusing and would want this to be intentional if it is included. 

When the two ids aren't null, are they typically the same value, or do they differ?

Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql:43]: fair, i think it should just be `fulfillment.location_id` then! that's actually where the location is stored as far as i can tell

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql]: updated!

File: models/intermediate/shopify__inventory_level__aggregates.sql
Comment (fivetran-jamie) [reply models/intermediate/shopify__inventory_level__aggregates.sql]: updated here and the source package!

File: DECISIONLOG.md
Comment (fivetran-joemarkiewicz) [inline DECISIONLOG.md:21]: Not sure what this piece is 🤔

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:69]: Will this be a separate update? If so, I can will skip over for this review.

File: dbt_project.yml
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:16]: Reminder to switch before merge

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie I have a few final questions about som `To Do` comments, but I believe these are probably for different updates to be completed outside of this PR.

If that is the case, the updates you made look great and I will approve this PR 😄

File: DECISIONLOG.md
Comment (fivetran-jamie) [reply DECISIONLOG.md:21]: note for later -- this is about no longer filtering out soft-deleted records and bringing in the is_Deleted fields (similar to the hubspot PR sheri just worked on)

File: README.md
Comment (fivetran-jamie) [reply README.md:69]: yes

**Pull request merged**
